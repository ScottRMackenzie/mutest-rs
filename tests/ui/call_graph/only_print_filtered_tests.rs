//@ print-call-graph
//@ stdout
//@ stderr: empty
//@ mutest-flags: --call-graph-filter-entry-points=test1

fn nested_common() {}

fn common() {
    nested_common();
}

fn indirectly_common() {}

fn nested_only_test_1() {
    indirectly_common();
}

fn only_test_1() {
    nested_only_test_1();
}

fn nested_only_test_2() {}

fn only_test_2() {
    nested_only_test_2();
    indirectly_common();
}

#[test]
fn test1() {
    common();
    only_test_1();
}

#[test]
fn test2() {
    common();
    only_test_2();
}
