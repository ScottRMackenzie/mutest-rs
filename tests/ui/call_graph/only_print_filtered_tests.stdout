entry points:

test test1
  -> common at tests/ui/call_graph/only_print_filtered_tests.rs:8:1: 8:12 (#0)
  -> only_test_1 at tests/ui/call_graph/only_print_filtered_tests.rs:18:1: 18:17 (#0)

nested calls at distance 1:

common at tests/ui/call_graph/only_print_filtered_tests.rs:8:1: 8:12 (#0)
  -> nested_common at tests/ui/call_graph/only_print_filtered_tests.rs:6:1: 6:19 (#0)
only_test_1 at tests/ui/call_graph/only_print_filtered_tests.rs:18:1: 18:17 (#0)
  -> nested_only_test_1 at tests/ui/call_graph/only_print_filtered_tests.rs:14:1: 14:24 (#0)

nested calls at distance 2:

nested_only_test_1 at tests/ui/call_graph/only_print_filtered_tests.rs:14:1: 14:24 (#0)
  -> indirectly_common at tests/ui/call_graph/only_print_filtered_tests.rs:12:1: 12:23 (#0)
